lnrpc/routerrpc/router.proto

```diff
@@ -776,6 +776,21 @@ message ForwardHtlcInterceptResponse {
 
     // The preimage in case the resolve action is Settle.
     bytes preimage = 3;
+
+    // (Encrypted) failure reason in case the resolve action is Fail. If left
+    // empty, a default temporary channel failure is returned to the sender.
+    bytes failure_reason = 4;
+
+    // Return UpdateFailMalformedHTLC in case the resolve action is Fail.
+    MalformedHtlcCode failure_malformed_htlc = 5;
```

Comment:
why did you choose to scope this to malformed htlc errors? As written, it would be easy to add other cases like `CodeChannelDisabled`, `CodeTemporaryChannelFailure`, `CodePermanentChannelFailure` which would be useful too.

Reply:
This is all that I needed for my purpose. It could be generalized, but then also all the failure message parameters need to be marshaled over rpc? Or would you do a hybrid where you only specify the code and lnd tries to fill in the message parameters?

Reply:
The ones I was thinking about are either very similar to existing cases or have no parameters. In any case, none of this needs to be implemented now. I was mostly wondering why the thight scope for the semantics.

Reply:
Reworked the PR to widen the scope.